homework/Task 2/DiagonalMatrix.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+
+namespace homework._Task2
+{
+    public class DiagonalMatrix : ICloneable
+    {
+        private readonly int _size;
+        private int[] _diagonal;
+
+        public int Size { get { return _size; } }
+        public int this[int x, int y]
+        {
+            get
+            {
+                if (x < Size &&  y < Size)
+                {
+                    return x == y? _diagonal[x] : 0;
+                }
+                return 0;
+            }
+            set
+            {
```

Comment:
The same as previous comment, x has to be equal to y, in other case its an incorrect operation